src/interpreter.ts

```diff
@@ -0,0 +1,176 @@
+import { Path, PathType } from './path';
+import { Code, GCodeCommand } from './gcode-parser';
+import { Machine } from './machine';
+
+export class Interpreter {
+  execute(commands: GCodeCommand[], machine = new Machine()): Machine {
+    commands.forEach((command) => {
+      if (command.code !== undefined) {
+        this[command.code](command, machine);
+      }
+    });
+
+    return machine;
+  }
+
+  G0(command: GCodeCommand, machine: Machine): void {
```

Comment:
Each method must be named after the enum for the command and accept those two params. There's probably a better way to do this.

Reply:
I can't think of any :-(

Reply:
Maybe it's a good thing to not be flexible?